### ISSUE TITLE
Example missing pthread linkage in cmake

### ISSUE THREAD

**Issue by Algomorph** [arpg/Node#19]
I'm not sure exactly what should be done, i.e. whether find_package(Threads) needs to be called somewhere in the CMakeLists of the Node or of the examples, but pthreads were missing for me on Ubuntu for the Node/examples.
I just added pthread to the linked for the examples, now it seems to build fine:
`target_link_libraries( NodeExample ${NODE_LIBRARIES} pthread)`

Before, I got: `undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'`

**Comment by crheckman** [comment]
This is a linux-only issue---we got it when we were building on Jenkins, too. Could you test #20, please?

**Comment by Algomorph** [comment]
The fix/linux_pthreads compiles fine for me, I think it's safe to merge. Not sure that just adding "pthread" is the 100% correct way to do it, though. Aren't other platforms also using threading?

**Comment by crheckman** [comment]
We don't explicitly use pthread anywhere in Node or NodeExample. GTest uses it, but while we've packaged some gtests in Node I don't see them implemented anywhere.

clang's relationship with the pthread flag is arcane (see tstack/lnav#153); it seems to require it when compiling a library which uses threads, but not when linking against the library that requires it. On the other hand, in the case that spawned this issue it appears pthread needs to be linked against explicitly.

I made an update to my comment above; Jenkins didn't have an issue with pthread before. We are running Ubuntu 14.04 on it with gcc version 4.8.2. What does your environment look like?

In #20 I was wondering whether this is a platform issue or a compiler issue, and it seems it's a compiler one, so I'll detect by compiler --- hopefully that fixes your problem.
